File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/cwe-190/code_before3.c
// From cwe-snippets, snippets_90/non-compliant/C/cwe-0190/char_fscanf_square_31.c

void func()
{
    char data;
    data = ' ';
    fscanf (stdin, "%c", &data);
    {
        char dataCopy = data;
        char data = dataCopy;
        {
            char result = data * data;
            printHexCharLine(result);
        }
    }
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/cwe-190/code_before3.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0190/short_rand_square_74b.java

public void func(HashMap<Integer,Short> dataHashMap ) throws Throwable
{
    short data = dataHashMap.get(2);

    short result = (short)(data * data);

    IO.writeLine("result: " + result);

}
